vault/plugin_catalog.go

```diff
@@ -41,78 +42,109 @@ type PluginCatalog struct {
 	logger          log.Logger
 
 	// multiplexedClients holds plugin process connections by plugin name
-	// This allows a single grpc connection to communicate with multiple
-	// databases. Each database configuration using the same plugin will be
-	// routed to the existing plugin process.
+	//
+	// This allows plugins that suppport multiplexing to use a single grpc
+	// connection to communicate with multiple "backends". Each backend
+	// configuration using the same plugin will be routed to the existing
+	// plugin process.
 	multiplexedClients map[string]*MultiplexedClient
 
 	lock sync.RWMutex
 }
 
+// PluginClient represents a connection to a plugin process
 type PluginClient struct {
```

Comment:
Do `PluginClient` and `MultiplexedClient` need to be exported?

Reply:
addressed in: https://github.com/hashicorp/vault/pull/13896/commits/c6a2e283f408fc0715f0fe2448699291ff93dc74